aws/resource_aws_s3outposts_endpoint.go

```diff
@@ -0,0 +1,212 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cidr_block": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"network_interfaces": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"network_interface_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"outpost_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"security_group_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func resourceAwsS3OutpostsEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).s3outpostsconn
+
+	input := &s3outposts.CreateEndpointInput{
+		OutpostId:       aws.String(d.Get("outpost_id").(string)),
+		SecurityGroupId: aws.String(d.Get("security_group_id").(string)),
+		SubnetId:        aws.String(d.Get("subnet_id").(string)),
+	}
+
+	output, err := conn.CreateEndpoint(input)
+
+	if err != nil {
+		return fmt.Errorf("error creating S3 Outposts Endpoint: %w", err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error creating S3 Outposts Endpoint: empty response")
+	}
+
+	d.SetId(aws.StringValue(output.EndpointArn))
+
+	if _, err := waiter.EndpointStatusCreated(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for S3 Outposts Endpoint (%s) to become available: %w", d.Id(), err)
+	}
+
+	return resourceAwsS3OutpostsEndpointRead(d, meta)
+}
+
+func resourceAwsS3OutpostsEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).s3outpostsconn
+
+	endpoint, err := finder.Endpoint(conn, d.Id())
+
+	if err != nil {
+		return fmt.Errorf("error reading S3 Outposts Endpoint (%s): %w", d.Id(), err)
+	}
+
+	if endpoint == nil {
+		if d.IsNewResource() {
+			return fmt.Errorf("error reading S3 Outposts Endpoint (%s): not found after creation", d.Id())
+		}
+
+		log.Printf("[WARN] S3 Outposts Endpoint (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", endpoint.EndpointArn)
+	d.Set("cidr_block", endpoint.CidrBlock)
+
+	if endpoint.CreationTime != nil {
+		d.Set("creation_time", aws.TimeValue(endpoint.CreationTime).Format(time.RFC3339))
+	}
+
+	if err := d.Set("network_interfaces", flattenS3outpostsNetworkInterfaces(endpoint.NetworkInterfaces)); err != nil {
+		return fmt.Errorf("error setting network_interfaces: %w", err)
+	}
+
+	d.Set("outpost_id", endpoint.OutpostsId)
```

Comment:
Outposts vs Outpost 🤷‍♀️

Reply:
The Create and Delete functions refer to it in the singular and recent ELBv2 support also refers to it in the singular: https://github.com/terraform-providers/terraform-provider-aws/pull/15170/files 🙁